src/main/java/de/tum/cit/aet/artemis/iris/service/session/IrisExerciseChatSessionService.java

```diff
@@ -134,19 +142,41 @@ public void checkRateLimit(User user) {
      * Sends all messages of the session to an LLM and handles the response by saving the message
      * and sending it to the student via the Websocket.
      *
-     * @param session The chat session to send to the LLM
+     * @param irisSession The chat session to send to the LLM. The full session is loaded from the database by ID.
      */
     @Override
-    public void requestAndHandleResponse(IrisExerciseChatSession session) {
-        var chatSession = (IrisExerciseChatSession) irisSessionRepository.findByIdWithMessagesAndContents(session.getId());
-        if (chatSession.getExercise().isExamExercise()) {
+    public void requestAndHandleResponse(IrisExerciseChatSession irisSession) {
+        var session = (IrisExerciseChatSession) irisSessionRepository.findByIdWithMessagesAndContents(irisSession.getId());
+        if (session.getExercise().isExamExercise()) {
             throw new ConflictException("Iris is not supported for exam exercises", "Iris", "irisExamExercise");
         }
-        var exercise = programmingExerciseRepository.findByIdWithTemplateAndSolutionParticipationElseThrow(chatSession.getExercise().getId());
-        var latestSubmission = getLatestSubmissionIfExists(exercise, chatSession.getUser());
 
-        var variant = irisSettingsService.getCombinedIrisSettingsFor(session.getExercise(), false).irisChatSettings().selectedVariant();
-        pyrisPipelineService.executeExerciseChatPipeline(variant, latestSubmission, exercise, chatSession);
```

Comment:
To be sure: You said you won't change anything functionality wise, but somehow remove this (and another) function. What am I missing, that it's not needed any more?

Reply:
Git shows the code as removed, but I actually just moved it. The `PyrisPipelineService` is used by all pipelines but contained some pipeline-specific code, now that code has been moved to the session services responsible for the specific pipelines. It looks a little different because `.executePipeline()` is now called directly from the session services instead of the helper methods, but I took care not to introduce any behavioral differences.